identity/src/main/java/com/android/identity/mdoc/mso/StaticAuthDataGenerator.java

```diff
@@ -0,0 +1,170 @@
+ * limitations under the License.
+ */
+
+package com.android.identity.mdoc.mso;
+
+import androidx.annotation.NonNull;
+
+import com.android.identity.internal.Util;
+
+import java.security.cert.X509Certificate;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.builder.ArrayBuilder;
+import co.nstant.in.cbor.builder.MapBuilder;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.SimpleValue;
+import co.nstant.in.cbor.model.SimpleValueType;
+import co.nstant.in.cbor.model.UnicodeString;
+
+/**
+ * Helper class for building <code>StaticAuthData</code> <a href="http://cbor.io/">CBOR</a> with
+ * the following CDDL:
+ * <pre>
+ *     StaticAuthData = {
+ *         "digestIdMapping": DigestIdMapping,
+ *         "issuerAuth" : IssuerAuth
+ *     }
+ *
+ *     DigestIdMapping = {
+ *         NameSpace =&gt; [ + IssuerSignedItemBytes ]
+ *     }
+ *
+ *     ; Defined in ISO 18013-5
+ *     ;
+ *     NameSpace = String
+ *     DataElementIdentifier = String
+ *     DigestID = uint
+ *     IssuerAuth = COSE_Sign1 ; The payload is MobileSecurityObjectBytes
+ *
+ *     IssuerSignedItemBytes = #6.24(bstr .cbor IssuerSignedItem)
+ *
+ *     IssuerSignedItem = {
+ *       "digestID" : uint,                           ; Digest ID for issuer data auth
+ *       "random" : bstr,                             ; Random value for issuer data auth
+ *       "elementIdentifier" : DataElementIdentifier, ; Data element identifier
+ *       "elementValue" : NULL                        ; Placeholder for Data element value
+ *     }
+ * </pre>
+ *
+ * <p>Note that the the byte[] arrays returned in the list of map values are
+ *  the bytes of IssuerSignedItem, not IssuerSignedItemBytes.
+ */
+public class StaticAuthDataGenerator {
+
+    private Map<String, List<byte[]>> mDigestIDMapping = new HashMap<>();
+    private byte[] mEncodedIssuerAuth;
+
+    /**
+     * Constructs a new {@link StaticAuthDataGenerator}.
+     */
+    public StaticAuthDataGenerator() {
+    }
+
+    /**
+     * Sets the <code>DigestIdMapping</code>. This must be called at least once before
+     * generating since <code>DigestIdMapping</code> must be non-empty.
+     *
+     * @param digestIDMapping A non-empty mapping between a <code>Namespace</code> and a list of
+     *                  <code>IssuerSignedItemBytes</code>.
+     * @return The <code>StaticAuthDataGenerator</code>.
+     * @exception IllegalArgumentException if the <code>digestIDMapping</code> is empty.
+     */
+    @NonNull
+    public StaticAuthDataGenerator setDigestIdMapping(@NonNull Map<String, List<byte[]>> digestIDMapping) {
+
+        if (digestIDMapping.isEmpty()) {
+            throw new IllegalArgumentException("digestIDs must not be empty");
+        }
+        mDigestIDMapping = digestIDMapping;
+        return this;
+    }
```

Comment:
If this must always be called probably cleaner to put it in the constructor? Same with `setEncodedIssuerAuth()` ...

Reply:
Agree